### PR TITLE
Homework 06

### PR DESCRIPTION
**Pull request by Fle9ma**
# Assignment
Until now, you were able to create cars with random manufacturers and models. In this homework, you will have to fetch and store car manufacturers and models in a database and use that information when creating a car to verify if manufacturer and model are valid.
You can fetch car manufacturers and models from [here](https://62d922dd5d893b27b2df0731.mockapi.io/api/v1/cars/1). Model your database entity to match data coming from the API and make sure you create a new table in a database where that data will be stored.
Create a scheduled task that will occasionally (e.g. once per day) fetch car manufacturers and models from the URL above and update the data in a database. When testing, you can change that value to something shorter (e.g. few seconds).

**IMPORTANT** Make sure car manufacturers and models data is not duplicated.

Replace `manufacturer` and `model` properties in the car class with one class that will combine that info and make sure database tables are linked (similar how car and car check-up are linked). Also, when creating a new car, validate manufacturer and model passed in a request using data stored in a database. Don’t forget to update tests to verify newly implemented logic.

**NOTE** You don’t have to test if scheduling works. This is part of the Spring and it’s already tested. Just make sure that logic triggered by a scheduled task works as expected.

# Optional
Use Redis/Caffeine to cache verification result when creating a new car. Invalidate cache when data is fetched from the API.
To test this, you will have to add Testcontainers support for Redis in your test or you can disable cache in tests.

### CODE REVIEW

**Comment by ivanmiklec** [comment]
Slobodno obriši par stvari koje više ne koristiš kao što su `kotlin/com/infinum/course/car/checkup/repository/DataRepository.kt` i neki exceptioni

**Comment by ivanmiklec** [comment]
Super izgleda PR 💪, prošao sam malo detaljnije ali nemam nekih ostalih zamjerki osim ovih par komentara koje sam ostavio

File: academy_project/src/main/kotlin/com/infinum/course/car/checkup/cardata/service/CarDataServiceImpl.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+@Service
+class CarDataServiceImpl(private val carDataRepository: CarDataRepository) : CarDataService {
+
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    @Cacheable("carData")
+    override fun getCarData(manufacturer: String, model: String): CarData? {
+        logger.info("Caching car data for manufacturer: $manufacturer and model: $model")
+        return carDataRepository.findByManufacturerAndModel(manufacturer, model)
```
[the remark this answers is not in the document]
Comment (Fle9ma) [reply academy_project/src/main/kotlin/com/infinum/course/car/checkup/cardata/service/CarDataServiceImpl.kt:17]: Ty 👍, yup puno bolje odmah to rijesiti nego provlaciti.

**Comment by Fle9ma** [comment]
> Slobodno obriši par stvari koje više ne koristiš kao što su `kotlin/com/infinum/course/car/checkup/repository/DataRepository.kt` i neki exceptioni

Ty 🙌 
Da li postoji mozda neki shortcut ili akcija koja se moze okinuti da ti kaze ako imas u projektu stvari koje vise ne koristis? 😊 Thanks

**Comment by ivanmiklec** [comment]
> Ty 🙌
Da li postoji mozda neki shortcut ili akcija koja se moze okinuti da ti kaze ako imas u projektu stvari koje vise ne koristis? 😊 Thanks

Totalno mi je promakao ovaj komentar u moru mailova, sori  😄. Ako koristiš Intellij:

1.  klikneš `shift` dva puta
2. upišeš `analyze code`
3. klikneš run inspection by name
4. upišeš unused declarations

**Review by ivanmiklec — APPROVED** [review]

**Comment by Fle9ma** [comment]
Ty na odgovoru, bio sam si kasnije na netu izvukao kako to postici 🙌

**Pull request merged**